controllers/release/adapter_test.go

```diff
@@ -782,6 +779,37 @@ var _ = Describe("Release adapter", Ordered, func() {
 			Expect(adapter.release.HasProcessingFinished()).To(BeTrue())
 		})
 
+		It("removes the finalizer if present", func() {
+			adapter.release.MarkProcessing("")
+
+			pipelineRun := &v1beta1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "pipeline-run",
+					Namespace:  "default",
+					Finalizers: []string{metadata.ReleaseFinalizer},
+				},
+			}
+			// The resource needs to be created as it will get patched
+			Expect(adapter.client.Create(adapter.ctx, pipelineRun)).To(Succeed())
+
+			pipelineRun.Status.MarkSucceeded("", "")
```

Comment:
if it is getting patched shouldn't this happen first? As in, before line 793?

Reply:
Nope. That only changes the status and a patch is not needed as we mock the return of the loader function and we don't rely on this status later on.

Reply:
so this is just so the HasProcessingFinished() statement hits?

Reply:
Yes.